src/nvim/channel.c

```diff
@@ -347,6 +348,7 @@ Channel *channel_job_start(char **argv, CallbackReader on_stdout,
     rstream_init(&proc->out, 0);
   }
 
+  chan->is_rpc = rpc;
```

Comment:
you can even remove this and change next if to `rpc`, `rpc_start` will set it to true.